scripts/tools/MonitoringExtraFieldConfigurationMigration.php

```diff
@@ -0,0 +1,115 @@
+use oat\taoProctoring\model\listener\MonitoringListenerInterface;
+use oat\taoProctoring\model\monitorCache\DeliveryMonitoringService;
+use oat\taoProctoring\model\repository\MonitoringRepository;
+use oat\taoQtiTest\models\event\QtiTestStateChangeEvent;
+use oat\taoTests\models\event\TestChangedEvent;
+use oat\taoTests\models\event\TestExecutionPausedEvent;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+class MonitoringExtraFieldConfigurationMigration extends ScriptAction implements ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+
+    protected function provideOptions()
+    {
+        return [];
+    }
+
+    protected function provideDescription()
+    {
+        return 'Migrate services and events to enable storage improvement for proctoring monitoring.';
+    }
+
+    protected function run()
+    {
+        $this->migrateEvents();;
+
+        $this->registerMonitoringRepository();
+
+        $report = new Report(Report::TYPE_SUCCESS, 'Events and services are now configured to use new data schema');
+        $report->add(
+            new Report(
+                Report::TYPE_INFO,
+                'If you have delivery executions, you should migrate them via: ' . PHP_EOL .
+                '`php index.php \'\oat\taoProctoring\scripts\tools\MonitoringExtraFieldMigration\' -h`'
+            )
+        );
+
+        return $report;
+    }
+
+    private function migrateEvents(): void
+    {
+        $eventManager = $this->getServiceLocator()->get(EventManager::SERVICE_ID);
+
+        $this->detachLegacyEvents($eventManager);
+        $this->attachNewEvents($eventManager);
+
+        $this->registerService(EventManager::SERVICE_ID, $eventManager);
+    }
+
+    private function detachLegacyEvents(EventManager $eventManager): void
+    {
+        $eventManager->detach(DeliveryExecutionState::class, [DeliveryMonitoringService::SERVICE_ID, 'executionStateChanged']);
+        $eventManager->detach(DeliveryExecutionCreated::class, [DeliveryMonitoringService::SERVICE_ID, 'executionCreated']);
+        $eventManager->detach(MetadataModified::class, [DeliveryMonitoringService::SERVICE_ID, 'deliveryLabelChanged']);
+        $eventManager->detach(TestChangedEvent::EVENT_NAME, [DeliveryMonitoringService::SERVICE_ID, 'testStateChanged']);
+        $eventManager->detach(QtiTestStateChangeEvent::EVENT_NAME, [DeliveryMonitoringService::SERVICE_ID, 'qtiTestStatusChanged']);
+        $eventManager->detach(AuthorizationGranted::EVENT_NAME, [DeliveryMonitoringService::SERVICE_ID, 'deliveryAuthorized']);
+        $eventManager->detach(TestExecutionPausedEvent::class, [DeliveryExecutionStateService::SERVICE_ID, 'catchSessionPause']);
+    }
+
+    private function attachNewEvents(EventManager $eventManager): void
+    {
+        $eventManager->attach(DeliveryExecutionState::class, [MonitoringListenerInterface::SERVICE_ID, 'executionStateChanged']);
```

Comment:
We have to register `MonitoringListenerInterface::SERVICE_ID` in SetrviceManager or do not use interface/SERVICE_ID at all and register it with class name. Since it doesn't have any configuration I think it would be better to use class name.
Right now EventManager cannot find this service `taoProctoring/MonitoringListener` and fails.